### DISCUSSION TITLE
Flow field inversion cannot be performed normally

### DISCUSSION THREAD

**Discussion by 15175829310** [mdolab/dafoam#673] [General]
[runScript_v2.txt](https://github.com/user-attachments/files/16755944/runScript_v2.txt)
I made modifications based on the example of flow field inversion and found that there were many things in the output file that should not have appeared, and there were errors when calculating the adjoint. I used periodic boundary conditions with a grid size of 172800 and 4 cores, but a single core cannot function properly. Below are my input and output files, and I hope you can help me find a solution
[logOpt.txt](https://github.com/user-attachments/files/16755941/logOpt.txt)

**Comment by 15175829310** [comment]
This issue has been resolved because I added average speed in the fvsolution file and fvsource in runscript_v2.py
But I don't know why this problem occurs. Can't I add source items to the fvsolution file?

**Comment by friedenhe** [comment]
Yeah, the fvSource is not supported well in DAFoam. If you want to add a source term, use DAFvSource instead.
